processing/src/main/java/io/druid/query/monomorphicprocessing/SpecializationService.java

```diff
@@ -335,11 +329,7 @@ private long addAndGetTotalIterationsOverTheLastHour(long newIterations)
         }
       }
       if (!currentMinutePresent) {
-        // putIfAbsent resolves a race between query processing threads calling accountLoopIterations() concurrently
-        AtomicLong existingValue = perMinuteIterations.putIfAbsent(currentMinute, new AtomicLong(newIterations));
-        if (existingValue != null) {
-          existingValue.addAndGet(newIterations);
-        }
+        perMinuteIterations.computeIfAbsent(currentMinute, AtomicLong::new).addAndGet(newIterations);
```

Comment:
This line actually contains bug, fixed here: https://github.com/druid-io/druid/pull/4079/commits/37b04c3984a4a2489ef65450028ec62e17db2d5a